class-inject/src/main/java/datadog/instrument/classinject/ClassInjector.java

```diff
@@ -165,74 +169,13 @@ public void visitCode() {
       mv.visitJumpInsn(IFEQ, notDatadogGlueRequest);
 
       if (JVM.atLeastJava(15)) {
-        // on Java 15+ prepare MethodHandles.lookup()
-        mv.visitMethodInsn(
-            INVOKESTATIC,
-            "java/lang/invoke/MethodHandles",
-            "lookup",
-            "()Ljava/lang/invoke/MethodHandles$Lookup;",
-            false);
-        mv.visitLdcInsn(DefineClassGlue.V9);
+        defineClassUsingMethodHandles();
```

Comment:
I'd prefer to keep the bytecode in a single method and use comments to document the flow, because it helps to visualize what it does.

Also the pieces are not meant to be re-used - if we did need to tweak the bytecode for a particular version then we'd have to un-duplicate the changes here first. In other words there are times when it is better to be verbose than attempt to de-duplicate as much as possible.

Reply:
Sure, I will drop the related commit